From 5495598f1a7ce03ed8cffcd13a5107410fc612c6 Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Tue, 02 Nov 2021 17:33:59 +0200
Subject: [PATCH] [SW-59942] greco: remove unwanted logs

Removed "Skipping halt for ARC" log in preboot.
Added log only if we need to halt ARC.
This is to make sure we are not cluttering the boot up logs

Change-Id: Iaa1686343dffa7d942ecf83cf3b3e4461df6a217
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index d0c180a..027e028 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -542,10 +542,9 @@
 	}
 
 	val = regmap_readl(halt_ack_offset);
-	if (val & halt_mask) {
-		preboot_puts_hex("Skipping halt for ARC", id, false);
-	} else {
-		/* Halt ARC */
+	if (!(val & halt_mask)) {
+		/* Halt ARC incase if it is not already */
+		preboot_puts_hex("Halting ARC", id, false);
 		regmap_writel(BIT(1), halt_req_offset);
 		while (1) {
 			val = regmap_readl(halt_ack_offset);
